runtime/basilisk/src/assets.rs

```diff
@@ -619,18 +636,61 @@ impl AmmTradeWeights<Trade<AssetId>> for RouterWeightInfo {
 			weight.saturating_accrue(amm_weight);
 		}
 
+		// Incorporate oracle price calculation
+		// We use xyk as reference
+		let weight_of_get_oracle_price_for_2_assets =
+			weights::pallet_route_executor::BasiliskWeight::<Runtime>::get_oracle_price_for_xyk();
+		let weight_of_get_oracle_price_for_route =
+			weight_of_get_oracle_price_for_2_assets.saturating_mul(route.len() as u64);
+		weight.saturating_accrue(weight_of_get_oracle_price_for_route);
+
 		weight
 	}
 
 	fn force_insert_route_weight() -> Weight {
 		//Since we don't have any AMM specific thing in the extrinsic, we just return the plain weight
 		weights::pallet_route_executor::BasiliskWeight::<Runtime>::force_insert_route()
 	}
+
+	// Used in OtcSettlements::settle_otc_order extrinsic
+	fn calculate_spot_price_with_fee_weight(route: &[Trade<AssetId>]) -> Weight {
+		let mut weight = Self::calculate_spot_price_overweight();
+
+		let lbp_weight = weights::pallet_lbp::BasiliskWeight::<Runtime>::calculate_spot_price_with_fee();
+		let xyk_weight = weights::pallet_xyk::BasiliskWeight::<Runtime>::calculate_spot_price_with_fee();
+
+		for trade in route {
+			let amm_weight = match trade.pool {
+				PoolType::LBP => lbp_weight,
+				PoolType::XYK => xyk_weight,
+				_ => lbp_weight.max(xyk_weight),
+			};
+			weight.saturating_accrue(amm_weight);
+		}
+
+		weight
+	}
+
+	fn get_route_weight() -> Weight {
+		weights::pallet_route_executor::BasiliskWeight::<Runtime>::get_route()
+	}
 }
 
 parameter_types! {
 	pub DefaultRoutePoolType: PoolType<AssetId> = PoolType::XYK;
+	pub const RouteValidationOraclePeriod: OraclePeriod = OraclePeriod::Hour;
```

Comment:
Because we don't use `TenMinutes` oracle in Basilisk: https://github.com/galacticcouncil/Basilisk-node/blob/27755941e2764bd6de1e3460ddd2ebef0c7410b0/runtime/basilisk/src/assets.rs#L648C15-L648C22